### PR TITLE
SCRUM-1711 API work for Disease Annotation creation

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurie.java [hunk 1]
```diff
@@ -27,6 +27,10 @@ public static String getConditionRelationUnique(ConditionRelation relation) {
 	}
 
 	public abstract String getCurieID(DiseaseAnnotationDTO annotationDTO);
+	
+	public abstract String getCurieID(GeneDiseaseAnnotation annotation);
+	public abstract String getCurieID(AlleleDiseaseAnnotation annotation);
+	public abstract String getCurieID(AGMDiseaseAnnotation annotation);
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurie.java]: @markquintontulloch curious why you need these three? Does the other function work its doing the same thing... there is a lot of duplicate code here?

File: src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/WormDiseaseAnnotationCurie.java [hunk 2]
```diff
@@ -28,4 +32,27 @@ public String getCurieID(String subject, String object, String reference, List<S
...
+	public String getCurieID(AlleleDiseaseAnnotation annotation) {
+		return getDiseaseAnnotationCurieID(annotation.getSubject().getCurie(), annotation);
+	}
+
+	@Override
+	public String getCurieID(AGMDiseaseAnnotation annotation) {
+		return getDiseaseAnnotationCurieID(annotation.getSubject().getCurie(), annotation);
+	}
+	
+	private String getDiseaseAnnotationCurieID(String subjectCurie, DiseaseAnnotation annotationDTO) {
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/WormDiseaseAnnotationCurie.java]: @markquintontulloch should probably pull DTO off the name here as its an entity and not a DTO.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/WormDiseaseAnnotationCurie.java]: done

File: src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurie.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurie.java]: We need the subject field to generate the curie, and that's not available from the core DiseaseAnnotation class - or am I missing a trick here?

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/diseaseAnnotations/DiseaseAnnotationCurie.java]: Yes thats true but every DiseaseAnnotation has a curie... putting in a PR to this so you can see what I mean.

**Comment by oblodgett** [comment]
@markquintontulloch did this PR mess up the IT's?

**Comment by markquintontulloch** [comment]
yep - `Unrecognized field "subjectCurie" (class org.alliancegenome.curation_api.model.entities.GeneDiseaseAnnotation), not marked as ignorable`

**Review by oblodgett — APPROVED** [review]

**Pull request merged**